test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibraryWithTfm.cs

```diff
@@ -0,0 +1,173 @@
+                new object[]
+                {
+                    "net40-client", null, true, false, false, false
+                },
+                new object[]
+                {
+                    "net45", null, true, false, false, false
+                },
+                new object[]
+                {
+                    "netstandard1.5", null, false, false, true, false
+                },
+                new object[]
+                {
+                    "portable-win81+wpa81", "$(MSBuildExtensionsPath)\\Microsoft\\Portable\\v4.6\\Microsoft.Portable.CSharp.targets", true, false, false, true
+                },
+                new object[]
+                {
+                    "portable-net451+wpa81+win81", "$(MSBuildExtensionsPath)\\Microsoft\\Portable\\v4.6\\Microsoft.Portable.CSharp.targets", true, false, false, false
+                },
+                new object[]
+                {
+                    "portable-net45+win8+wp8+wpa81", "$(MSBuildExtensionsPath)\\Microsoft\\Portable\\v4.5\\Microsoft.Portable.CSharp.targets", true, false, false, false
+                },
+                new object[]
+                {
+                    "portable-net4+sl5+win8+wpa81+wp8", "$(MSBuildExtensionsPath)\\Microsoft\\Portable\\v4.0\\Microsoft.Portable.CSharp.targets", true, false, false, false
+                },
+                new object[]
+                {
+                    "uap10.0", "$(MSBuildExtensionsPath)\\Microsoft\\WindowsXaml\\v15.0\\Microsoft.Windows.UI.Xaml.CSharp.targets", false, false, false, true
+                },
+                new object[]
+                {
+                    "xamarinios", "$(MSBuildExtensionsPath)\\Xamarin\\iOS\\Xamarin.iOS.CSharp.targets", false, true, false, false
+                },
+                new object[]
+                {
+                    "xamarinmac", "$(MSBuildExtensionsPath)\\Xamarin\\Mac\\Xamarin.Mac.CSharp.targets", false, false, false, false
+                },
+                new object[]
+                {
+                    "xamarintvos", "$(MSBuildExtensionsPath)\\Xamarin\\TVOS\\Xamarin.TVOS.CSharp.targets", false, true, false, false
+                },
+                new object[]
+                {
+                    "xamarinwatchos", "$(MSBuildExtensionsPath)\\Xamarin\\WatchOS\\Xamarin.WatchOS.CSharp.targets", false, true, false, false
+                },
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestData))]
+        public void It_builds_the_tfm_library_successfully_on_windows(string tfm, string langTargets, 
+                                                                      bool hasLocalRef, bool hasMdb, bool hasDeps, bool hasPri)
+        {
+            if (!UsingFullFrameworkMSBuild)
+            {
+                return;
+            }
+            
+            var targetsExist = TargetsExist(langTargets);
+            if (!targetsExist)
+                return; // Generate warning
```

Comment:
Is this a TODO?  I would suggest the behavior here be to silently skip a test where targets don't exist, unless an option (e.g. an environment variable) was set to force all tests to be run, in which case the test would fail.